src/main/java/hudson/remoting/Channel.java

```diff
@@ -1030,21 +1031,29 @@ public void terminate(@Nonnull IOException e) {
                     logger.log(Level.WARNING, "Failed to close down the reader side of the transport", x);
                 }
                 try {
+                    logger.log(Level.WARNING, "Clearing pending calls");
                     synchronized (pendingCalls) {
                         for (Request<?, ?> req : pendingCalls.values())
                             req.abort(e);
                         pendingCalls.clear();
                     }
+                    logger.log(Level.WARNING, "Clearing executing calls");
                     synchronized (executingCalls) {
                         for (Request<?, ?> r : executingCalls.values()) {
                             java.util.concurrent.Future<?> f = r.future;
                             if (f != null) f.cancel(true);
                         }
                         executingCalls.clear();
                     }
+                    logger.log(Level.WARNING, "Aborting exported objects");
                     exportedObjects.abort(e);
                     // break any object cycles into simple chains to simplify work for the garbage collector
+                    logger.log(Level.WARNING, "Clearing reference");
                     reference.clear(e);
+                    logger.log(Level.WARNING, "Successfully terminated channel, still need to notify listeners");
+                } catch (Throwable t) {
+                    logger.log(Level.WARNING, "Caught error while terminating channel", t);
```

Comment:
Caught **unhandled** error. Maybe it should be severe since it may be fatal to the upstream logic

Reply:
Ah whoops, forgot to commit the change where I rethrew it.  Thanks for the catch.